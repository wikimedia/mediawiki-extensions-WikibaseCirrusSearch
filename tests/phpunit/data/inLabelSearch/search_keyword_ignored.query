{
	"search": "haswbstatement:P31=Q5 Jane Doe",
	"language": "en",
	"type": "item"
}
