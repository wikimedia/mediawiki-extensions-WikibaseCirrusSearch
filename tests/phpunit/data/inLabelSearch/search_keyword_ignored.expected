{
    "description": "wikibase_in_label search for 'haswbstatement:P31=Q5 Jane Doe'",
    "params": {
        "timeout": "20s"
    },
    "query": {
        "query": {
            "bool": {
                "minimum_should_match": 1,
                "filter": [
                    {
                        "term": {
                            "content_model": "wikibase-item"
                        }
                    }
                ],
                "should": [
                    {
                        "bool": {
                            "filter": [
                                {
                                    "match": {
                                        "labels_all.plain": {
                                            "query": "haswbstatement:P31=Q5 Jane Doe",
                                            "operator": "and"
                                        }
                                    }
                                }
                            ],
                            "should": [
                                {
                                    "dis_max": {
                                        "tie_breaker": 0,
                                        "queries": [
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.en.near_match": "haswbstatement:P31=Q5 Jane Doe"
                                                        }
                                                    },
                                                    "boost": 2
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.en.near_match_folded": "haswbstatement:P31=Q5 Jane Doe"
                                                        }
                                                    },
                                                    "boost": 1.6
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.en.plain": "haswbstatement:P31=Q5 Jane Doe"
                                                        }
                                                    },
                                                    "boost": 1.1
                                                }
                                            }
                                        ]
                                    }
                                }
                            ]
                        }
                    },
                    {
                        "term": {
                            "title.keyword": "HASWBSTATEMENT:P31=Q5 JANE DOE"
                        }
                    }
                ]
            }
        },
        "_source": [
            "namespace",
            "title",
            "namespace_text",
            "wiki",
            "labels.en",
            "descriptions.en"
        ],
        "fields": [],
        "highlight": {
            "pre_tags": [
                ""
            ],
            "post_tags": [
                ""
            ],
            "fields": {
                "title": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "number_of_fragments": 0,
                    "matched_fields": [
                        "title.keyword"
                    ]
                },
                "labels.en.plain": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "number_of_fragments": 0,
                    "options": {
                        "skip_if_last_matched": true,
                        "return_snippets_and_offsets": true
                    }
                },
                "labels.*.plain": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "number_of_fragments": 0,
                    "options": {
                        "skip_if_last_matched": true,
                        "return_snippets_and_offsets": true
                    }
                }
            }
        },
        "size": 10,
        "rescore": [
            {
                "window_size": 8192,
                "query": {
                    "query_weight": 1,
                    "rescore_query_weight": 1,
                    "score_mode": "total",
                    "rescore_query": {
                        "function_score": {
                            "score_mode": "sum",
                            "functions": [
                                {
                                    "script_score": {
                                        "script": {
                                            "source": "pow(doc['incoming_links'].value , 1) \/ ( pow(doc['incoming_links'].value, 1) + pow(100,1))",
                                            "lang": "expression"
                                        }
                                    },
                                    "weight": 0.6
                                },
                                {
                                    "script_score": {
                                        "script": {
                                            "source": "pow(doc['sitelink_count'].value , 2) \/ ( pow(doc['sitelink_count'].value, 2) + pow(20,2))",
                                            "lang": "expression"
                                        }
                                    },
                                    "weight": 0.4
                                }
                            ]
                        }
                    }
                }
            }
        ],
        "stats": [
            "wikibase-in-label"
        ]
    },
    "options": {
        "timeout": "20s"
    }
}